### PR TITLE
Minor bug fixes, add PR test workflow

### PR DESCRIPTION
**Pull request by mprimi**

### CODE REVIEW

File: pkg/sys/api_test.go
Comment (piotrpio) [inline pkg/sys/api_test.go:1]: This does not compile, you should probably run `go fmt` to remove unused imports.

**Review by piotrpio — COMMENTED** [review]
CI did not trigger for some reason - let's see if it will work after you push new commit.

File: pkg/sys/api_test.go
Comment (mprimi) [reply pkg/sys/api_test.go:1]: Fixed. The package structure is a little weird so `gofmt` didn't even see those unused imports

Comment (mprimi) [reply pkg/sys/api_test.go:1]: And CI will take the workflow only after is merged into main for security reasons.

Comment (mprimi) [reply pkg/sys/api_test.go:1]: Nevermind, looks like it's running. That failing test is flaky on my machine too.

**Review by piotrpio — APPROVED** [review]
LGTM!

**Pull request merged**